### PR TITLE
1692 integration erfassung urneninhalt in anwendung

### LINKED ISSUE #1692: Integration Erfassung Urneninhalt in Anwendung
# Vorausetzungen
- #1690
  - Komponenten die je nach Wahlbezirkart die korrekten Felder anzeigt
- #1688
  - Der Grundaufbau des Navigation für die Ergebnisermittelung wurde erstellt

# Umsetzungshinweise
- die View erhält über die URL die wahlID
  - `.../<Name der View>/{wahlID}`
  - `wahlID` via Watcher überwachen um auf Änderungen reagieren zu können
- Verwendung der Komponente aus #1690 in der View
  - Titel bei UWB: Wahlurne öffnen und Stimmzettel zählen 
    - useTime ist false
  - Titel bei BWB: Wahlurne öffnen und Stimmzettelumschläge zählen
    - iseTime ist true
- Zur Einordnung in die Navigation in die Altanwendung schauen
  - die Steps wurden zu Untereinträgen bei der Navigation

# Akzeptanzkriterien
- [ ] Es gibt für jede Wahl einen Navigationseintrag der zur der View führt um die Daten zu erfassen
  - dieser ist an erster Stelle
  - Bezeichnung bei UWB: Zählen der Stimmzettel
  - Bezeichnung bei BWB: Zählen der Stimmzettelumschläge
- [ ] nach der korrekten Erfassen sind die Daten bei der entsprechenden Wahl gespeichert
- [ ] der UseCase ist in der Fachlichen Doku beschrieben

### PR DESCRIPTION
**Pull request by vjohnslhm**
# Beschreibung:

erfassung der stimmzettelumschläge integriert

### Dokumentation
- [x] Links geprüft
- [x] Texte auf Rechtschreibung und Grammatik geprüft

### Frontend
- [x] [Naming Conventions][naming-conventions-link] beachtet
- [x] Doku aktualisiert
  - [Fachliche Beschreibung][fachliche-beschreibung-link]
- [x] Unit-Tests gepflegt
- [x] Komponententests gepflegt
- [x] Texte auf Rechtschreibung und Grammatik geprüft

# Referenzen[^1]:

Closes #1692

> [^1]: _Nicht zutreffende Referenzen vor dem Speichern entfernen_

[naming-conventions-link]: https://it-at-m.github.io/Wahllokalsystem/technik/naming_conventions/
[fachliche-beschreibung-link]: https://it-at-m.github.io/Wahllokalsystem/about/#fachliche-anforderungen

## Summary by CodeRabbit

- New Features
  - Added a dedicated “Stimmzettel” counting view with dynamic title and time-tracking behavior based on user role.
  - Introduced a new route for the counting view, supporting optional election ID navigation.
  - Updated navigation items to link directly to the counting view for relevant election types, passing the appropriate election ID automatically.
  - Improved routing behavior to keep the URL in sync when the selected election changes.

- Tests
  - Added tests covering election ID resolution by election type and edge cases.

### CODE REVIEW

File: wls-gui-wahllokalsystem/src/views/auszaehlung/ErfassungStimmzettelView.vue [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+import TheErgebnisermittlungStimmzettelumschlaegeCard from "@/components/ergebnisermittlung/TheErgebnisermittlungStimmzettelumschlaegeCard.vue";
+import { ROUTE_AUSZAEHLUNG_STIMMZETTEL } from "@/constants.ts";
+import { useUserStore } from "@/stores/userStore.ts";
+import { useWahlenStore } from "@/stores/wahlenStore.ts";
+
+const route = useRoute();
+const router = useRouter();
+const { isUWB } = useUserStore();
+const { getWahlOrUndefinedById } = useWahlenStore();
```
[the remark this answers is not in the document]
Comment (vjohnslhm) [reply wls-gui-wahllokalsystem/src/views/auszaehlung/ErfassungStimmzettelView.vue]: [497fc81](https://github.com/it-at-m/Wahllokalsystem/pull/1802/commits/497fc814e73afcb808530a1382c6ac12f8d2485a)

File: wls-gui-wahllokalsystem/src/components/navigation/auszaehlung_wahlarten/TheSRWScoresListGroup.vue [hunk 2]
```diff
@@ -6,11 +6,34 @@
...
+    <!-- [Vue warn]: Vuetify error: Multiple nodes with the same ID ist ein bekannter vuetify bug und kommt daher,
+    dass das list-item für BAW und OBW mit der gleichen Route aufgerufen wird. Siehe
+    https://github.com/vuetifyjs/vuetify/issues/20516 -->
+    <v-list-item
+      title="Zählen der Stimmzettel"
+      :to="{
+        name: ROUTE_AUSZAEHLUNG_STIMMZETTEL,
+        params: { wahlId: srwWahlID },
+      }"
+    />
```
[the remark this answers is not in the document]
Comment (vjohnslhm) [reply wls-gui-wahllokalsystem/src/components/navigation/auszaehlung_wahlarten/TheSRWScoresListGroup.vue]: [497fc81](https://github.com/it-at-m/Wahllokalsystem/pull/1802/commits/497fc814e73afcb808530a1382c6ac12f8d2485a)

File: wls-gui-wahllokalsystem/src/components/navigation/auszaehlung_wahlarten/TheBAWScoresListGroup.vue [hunk 3]
```diff
@@ -6,11 +6,34 @@
...
+    <!-- [Vue warn]: Vuetify error: Multiple nodes with the same ID ist ein bekannter vuetify bug und kommt daher,
+    dass das list-item für SRW und OBW mit der gleichen Route aufgerufen wird. Siehe
+    https://github.com/vuetifyjs/vuetify/issues/20516 -->
+    <v-list-item
+      title="Zählen der Stimmzettel"
+      :to="{
+        name: ROUTE_AUSZAEHLUNG_STIMMZETTEL,
+        params: { wahlId: bawWahlID },
+      }"
+    />
```
[the remark this answers is not in the document]
Comment (vjohnslhm) [reply wls-gui-wahllokalsystem/src/components/navigation/auszaehlung_wahlarten/TheBAWScoresListGroup.vue]: [497fc81](https://github.com/it-at-m/Wahllokalsystem/pull/1802/commits/497fc814e73afcb808530a1382c6ac12f8d2485a)

File: wls-gui-wahllokalsystem/src/stores/wahlenStore.ts [hunk 4]
```diff
@@ -189,6 +190,14 @@ export const useWahlenStore = defineStore(storeID, () => {
...
   }
 
+  function getWahlIdOrUndefinedByWahlart(wahlart: WahlWahlartEnum) {
+    if (wahlen.value) {
+      const wahl = wahlen.value.find((wahl) => wahl.wahlart == wahlart);
+
+      return wahl ? wahl.wahlID : undefined;
+    }
+  }
+
```
[the remark this answers is not in the document]
Comment (vjohnslhm) [reply wls-gui-wahllokalsystem/src/stores/wahlenStore.ts]: [8a9d7f6](https://github.com/it-at-m/Wahllokalsystem/pull/1802/commits/8a9d7f6ce92aaceaf5dd25241c4d95a5ea10a02e)

File: wls-gui-wahllokalsystem/src/components/navigation/auszaehlung_wahlarten/TheBAWScoresListGroup.vue [hunk 5]
```diff
@@ -6,11 +6,43 @@
...
+import { WahlWahlartEnum } from "@/types/wahl/WahlWahlartEnum.ts";
+
+const { getWahlIdOrUndefinedByWahlart } = useWahlenStore();
+const { isUWB } = storeToRefs(useUserStore());
+
+const title = computed(() => {
+  return isUWB.value
+    ? "Zählen der Stimmzettel"
+    : "Zählen der Stimmzettelumschläge";
+});
```
Comment (MrSebastian) [inline wls-gui-wahllokalsystem/src/components/navigation/auszaehlung_wahlarten/TheBAWScoresListGroup.vue]: Die Logik wird in alle 3 Listen verwendet. Da es die selbe Logik sein soll bitte auslagern damit es nur an einer Stelle angepasst werden muss.

File: wls-gui-wahllokalsystem/src/views/auszaehlung/ErfassungStimmzettelView.vue [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  />
+</template>
+
+<script setup lang="ts">
+import { storeToRefs } from "pinia";
+import { computed, watch } from "vue";
+import { useRoute, useRouter } from "vue-router";
+
+import TheErgebnisermittlungStimmzettelumschlaegeCard from "@/components/ergebnisermittlung/TheErgebnisermittlungStimmzettelumschlaegeCard.vue";
+import { ROUTE_AUSZAEHLUNG_STIMMZETTEL } from "@/constants.ts";
```
Comment (MrSebastian) [inline wls-gui-wahllokalsystem/src/views/auszaehlung/ErfassungStimmzettelView.vue]:
```suggestion
import { EXAMPLE_ROUTES_NOTFOUND} from "@/constants.ts";
```

File: wls-gui-wahllokalsystem/src/views/auszaehlung/ErfassungStimmzettelView.vue [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  (newId) => {
+    const wahl = getWahlOrUndefinedById(newId as string);
+
+    if (wahl) {
+      router.push({
+        name: ROUTE_AUSZAEHLUNG_STIMMZETTEL,
+        params: { wahlId: String(wahl.wahlID) },
+      });
+    }
+  }
```
Comment (MrSebastian) [inline wls-gui-wahllokalsystem/src/views/auszaehlung/ErfassungStimmzettelView.vue]: Wenn es zu der ID keine Wahl gibt würde ich auf die Error-Seite gehen

```suggestion
  () => wahlID.value,
  (wahlID) => {
    const wahl = getWahlOrUndefinedById(wahlID);

    if (!wahl) {
      router.push({
        name: EXAMPLE_ROUTES_NOTFOUND,
      });
    }
  }
```

**Review by MrSebastian — CHANGES_REQUESTED** [review]

File: wls-gui-wahllokalsystem/src/views/auszaehlung/ErfassungStimmzettelView.vue (hunk 7 above)
Comment (vjohnslhm) [reply wls-gui-wahllokalsystem/src/views/auszaehlung/ErfassungStimmzettelView.vue]: [06d3f50](https://github.com/it-at-m/Wahllokalsystem/pull/1802/commits/06d3f50aa18e6140a07efc27ec425c98b9f503b8)

File: wls-gui-wahllokalsystem/src/views/auszaehlung/ErfassungStimmzettelView.vue (hunk 6 above)
Comment (vjohnslhm) [reply wls-gui-wahllokalsystem/src/views/auszaehlung/ErfassungStimmzettelView.vue]: [06d3f50](https://github.com/it-at-m/Wahllokalsystem/pull/1802/commits/06d3f50aa18e6140a07efc27ec425c98b9f503b8)

File: wls-gui-wahllokalsystem/src/components/navigation/auszaehlung_wahlarten/TheBAWScoresListGroup.vue (hunk 5 above)
Comment (vjohnslhm) [reply wls-gui-wahllokalsystem/src/components/navigation/auszaehlung_wahlarten/TheBAWScoresListGroup.vue]: [0597c32](https://github.com/it-at-m/Wahllokalsystem/pull/1802/commits/0597c32ddea0c751fccb980ae018967580507db2)

**Review by MrSebastian — APPROVED** [review]
